### ISSUE TITLE
Performance regression: replacing int by size_t (while moving npar to CPPProcess.h) increases the number of registers

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#669]
Intially "Understand why moving npar to CPPProcess.h increases the number of registers" [later on: undertood this is int vs size_t]

This is a follow-up to MR #668, which I will probably merge in any case before understanding/fixing this issue

To support processes with P1 and P2 with differnt numbers of particles (e.g gg to tt or ttg), some parameters like npar must be moved from src to P1 (issue #667, addressed in MR #668). However this leads to a higher number of registers in CUDA. This is to be understood.

Note: unfortunately I had the bad idea of mixing this with a change from int to size_t for the parameters. I wonder if this is the cause of the issue?

**Comment by valassi** [comment]
For the performane regression, see this: reverting from new to old code decreases registers from 164 to 150

https://github.com/madgraph5/madgraph4gpu/pull/668/commits/29127bb09a3608fd418a9c8fa20fded3a231539c

```
 runNcu /data/avalassi/GPU2023/madgraph4gpuX/epochX/cudacpp/ee_mumu.mad/SubProcesses/P1_epem_mupmum/build.none_d_inl0_hrd0/gcheck.exe -p 2048 256 1
-==PROF== Profiling "sigmaKin": launch__registers_per_thread 164
+==PROF== Profiling "sigmaKin": launch__registers_per_thread 150
```

**Comment by valassi** [comment]
I understood what causes this regression: it is my move to size_t from int, it is NOT the fault of the move from mgOnGpuConfig.h to CPPProcess.h

**I will move back from size_t to int for these parameters**... I also opened #670 for an eventual cleanup and more general study

**Comment by valassi** [comment]
This is fixed in MR #668 - it can be closed. Further studies will be in #670
